addons/account/models/account_journal_dashboard.py

```diff
@@ -102,6 +104,33 @@ def get_line_graph_datas(self):
         color = '#875A7B' if '+e' in version else '#7c7bad'
         return [{'values': data, 'title': graph_title, 'key': graph_key, 'area': True, 'color': color}]
 
+    # Below method is used to get data of bank statemens
+    @api.multi
+    def get_bank_stmt_line_graph_datas(self):
+        data = []
+        Statement = self.env['account.bank.statement']
+        StatementLine = self.env['account.bank.statement.line']
+        today = datetime.today()
+        locale = self._context.get('lang') or 'en_US'
+        show_date = today + timedelta(days=-30)
+        start_balance = 0.0
+        for date_range in range(0, 7):
+            if show_date <= today:
+                statement = Statement.search([('journal_id', 'in', self.ids), ('date', '<', show_date.strftime(DF))], order="date desc", limit=1)
+                if not statement:
+                    lines = StatementLine.search([('journal_id', 'in', self.ids), ('date', '<=', show_date.strftime(DF))])
```

Comment:
I think `('date', '>', ...)` is missing here.

Suppose 3 dates d0, d1, d2 with d0 < d1 < d2 and d2 - d0 >= 15 days, d1 - d0 >= 5 days.
d0, d1 are statement lines dates and d2 the date of the statement.

First iteration:
The line with d0 date is considered because no statement found.

Second iteration:
The lines corresponding to d0 and d1 are considered because no statement found.

Problem:
The amount of the line at d0 is used twice.

Reply:
we are using sum of all the bank statement lines which are smaller than shown date on the graph, so there is no need to compare date > operation, as we are summing up the amount of bank statement lines, for further explanation please check the attached video.

link - https://drive.google.com/file/d/0B_wmYnbN259hYXZMLXAtdGdUTkU/view